File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum : int = 0
        maxi : int = -sys.maxsize - 1

        for i in range (0,len(nums)):
            sum += nums[i]

            if(sum > maxi):
                maxi = sum
            
            if (sum < 0):
                sum = 0

        return maxi
        

File: 53-maximum-subarray/maximum-subarray.ts
function maxSubArray(nums: number[]): number {
    let sum : number = 0;
    let maxi : number = Number.MIN_SAFE_INTEGER;

    for(let i:number = 0;i<nums.length;i++)
    {
        sum += nums[i];

        if(sum > maxi)
        {
            maxi = sum;
        }

        if(sum < 0)
        {
            sum = 0;
        }
    }
    return maxi;
};